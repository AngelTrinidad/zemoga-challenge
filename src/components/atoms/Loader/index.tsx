import React, {FC} from 'react';
import {ActivityIndicator} from 'react-native';

import {useTheme} from '~contexts';

interface Props {
  isLoading?: boolean;
  size?: 'small' | 'large';
  color?: string;
}

const Loader: FC<Props> = ({color, isLoading = true, size = 'large'}) => {
  const {colors} = useTheme();

  return (
    <ActivityIndicator
      color={color || colors.primary}
      size={size}
      animating={isLoading}
    />
  );
};

export default Loader;
