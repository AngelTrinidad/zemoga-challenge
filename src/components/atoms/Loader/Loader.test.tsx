import React from 'react';

import Loader from '.';

import {render} from '~helpers/testing';

describe('<Loader />', () => {
  let component: any;

  beforeEach(() => {
    component = render(<Loader />);
  });

  it('should render correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
