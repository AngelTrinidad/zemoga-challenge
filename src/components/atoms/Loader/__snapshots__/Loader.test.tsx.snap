// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should has same snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="#567189"
  size="large"
/>
`;
